internal/setup/setup.go

```diff
@@ -542,14 +572,36 @@ func parseRelease(baseDir, filePath string, data []byte) (*Release, error) {
 			}
 			archiveKeys = append(archiveKeys, key)
 		}
+		if details.Priority > MaxArchivePriority || details.Priority < MinArchivePriority {
+			return nil, fmt.Errorf("%s: archive %q has invalid priority value of %d", fileName, archiveName, details.Priority)
+		}
+		if details.Priority != 0 {
+			hasPriority = true
```

Comment:
The handling of priority zero seems a bit loose at the moment. We consider it not being zero here as "it has priority", but then right below we actually define the default (!) priority as zero.  Also, we error if two priorities are the same, which forces people to enter a priority value for all but one of them, which will be automatically set to zero due to how yaml unmarshaling takes place on the int type.

It looks like we might fix all of these issues at once by forcing the priority, if present, to be either positive or negative, and considering zero to be unset per the test above. On unmarshalling, if we can clarify the situation further by having the Priority field on the yaml type being a pointer, so we can differentiate between unset and zero, and then failing on inconsistency (any priority is set but some not, or no default and no priority).

Reply:
Done in https://github.com/canonical/chisel/pull/160/commits/418ea0d85a5d307da5801536b9ab98d1af65fb55, I have changed the priority to be a pointer so we can distinguish when it is 0 and when it is unset. Also, we now return an error in the case there is an inconsistent use of priorities. The only bit I am not sure about is the error messages when there are no priorities set across several archives (I have included a test in the commit to show that).

Reply:
I think the error we want in those cases is to point out just one of them as a hint. Something like:
```
error: archive %q is missing the priority setting
```
We can raise this when there is no `defaultArchive`, and we have a `priorityArchive` (which needs support, equivalent to `noPriorityArchive`).

Reply:
Done. We know check if there is more than one archive and there is no default and there are no priorities.